### PR TITLE
feat: add FileListHeader

### PR DESCRIPTION
**Pull request by skjnldsv**
Ref: https://github.com/nextcloud/server/pull/17482

It seems no one is using the footer API in years. I think it's time to drop it :shrug:

### CODE REVIEW

File: lib/fileListHeaders.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import { Folder } from './files/folder'
+import logger from './utils/logger'
+
+export interface HeaderData {
+	/** Unique ID */
+	id: string
+	/** Order */
+	order: number
+	/** Condition wether this header is shown or not */
+	enabled?: (folder: Folder, view) => boolean
```
Comment (juliushaertl) [inline lib/fileListHeaders.ts:32]: Should the view be typed somehow?

**Review by juliushaertl — APPROVED** [review]
Looks good, just a small question on the view parameter typing.

File: lib/fileListHeaders.ts (hunk 1 above)
Comment (susnux) [reply lib/fileListHeaders.ts:32]: I agree we should have the view types (if possible)

**Review by susnux — APPROVED** [review]
Same as Julius just the view typing

File: lib/fileListHeaders.ts (hunk 1 above)
Comment (skjnldsv) [reply lib/fileListHeaders.ts:32]: Yes, we haven't migrated it from server yet! Was maing sure no changes were needed (I had to adjust quite a lot at the beginning)

**Pull request merged**